src/moj/components/date-picker/date-picker.js

```diff
@@ -0,0 +1,948 @@
+  this.$dialog.style.display = "none";
+  this.$dialog.classList.remove("moj-datepicker-dialog--open");
+  this.$calendarButton.setAttribute("aria-expanded", "false");
+  this.$calendarButton.focus();
+};
+
+Datepicker.prototype.goToDate = function (date, focus) {
+  const current = this.currentDate;
+  this.currentDate = date;
+
+  if (
+    current.getMonth() !== this.currentDate.getMonth() ||
+    current.getFullYear() !== this.currentDate.getFullYear()
+  ) {
+    this.updateCalendar();
+  }
+
+  this.setCurrentDate(focus);
+};
+
+// day navigation
+Datepicker.prototype.focusNextDay = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() + 1);
+  this.goToDate(date);
+};
+
+Datepicker.prototype.focusPreviousDay = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() - 1);
+  this.goToDate(date);
+};
+
+// week navigation
+Datepicker.prototype.focusNextWeek = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() + 7);
+  this.goToDate(date);
+};
+
+Datepicker.prototype.focusPreviousWeek = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() - 7);
+  this.goToDate(date);
+};
+
+Datepicker.prototype.focusFirstDayOfWeek = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() - date.getDay());
+  this.goToDate(date);
+};
+
+Datepicker.prototype.focusLastDayOfWeek = function () {
+  const date = new Date(this.currentDate);
+  date.setDate(date.getDate() - date.getDay() + 6);
+  this.goToDate(date);
+};
+
+// month navigation
+Datepicker.prototype.focusNextMonth = function (event, focus = true) {
+  event.preventDefault();
+  const date = new Date(this.currentDate);
+  date.setMonth(date.getMonth() + 1, 1);
+  this.goToDate(date, focus);
+};
+
+Datepicker.prototype.focusPreviousMonth = function (event, focus = true) {
+  event.preventDefault();
+  const date = new Date(this.currentDate);
+  date.setMonth(date.getMonth() - 1, 1);
+  this.goToDate(date, focus);
+};
+
+// year navigation
+Datepicker.prototype.focusNextYear = function (event, focus = true) {
+  event.preventDefault();
+  const date = new Date(this.currentDate);
+  date.setFullYear(date.getFullYear() + 1, date.getMonth(), 1);
+  this.goToDate(date, focus);
+};
+
+Datepicker.prototype.focusPreviousYear = function (event, focus = true) {
+  event.preventDefault();
+  const date = new Date(this.currentDate);
+  date.setFullYear(date.getFullYear() - 1, date.getMonth(), 1);
+  this.goToDate(date, focus);
+};
+
+/**
+ * Parse dataset
+ *
+ * Loop over an object and normalise each value using {@link normaliseString},
+ * optionally expanding nested `i18n.field`
+ *
+ * @param {{ schema: Schema }} Component - Component class
+ * @param {DOMStringMap} dataset - HTML element dataset
+ * @returns {Object} Normalised dataset
+ */
+Datepicker.prototype.parseDataset = function (schema, dataset) {
```

Comment:
Potentially early optimisation, but might be worth moving this and `mergeConfigs` into helper functions since they'll probably be used by other components in the future too